homeassistant/helpers/condition.py

```diff
@@ -227,16 +227,25 @@ async def async_from_config(
         factory = platform.async_condition_from_config
 
     # Check if condition is not enabled
-    if not config.get(CONF_ENABLED, True):
-
-        @trace_condition_function
-        def disabled_condition(
-            hass: HomeAssistant, variables: TemplateVarsType = None
-        ) -> bool | None:
-            """Condition not enabled, will act as if it didn't exist."""
-            return None
-
-        return disabled_condition
+    if CONF_ENABLED in config:
+        enabled = config[CONF_ENABLED]
+        if isinstance(enabled, Template):
+            try:
+                enabled = enabled.async_render(limited=True)
+            except TemplateError as err:
+                raise HomeAssistantError(
+                    f"Error rendering condition enabled template: {err}"
+                ) from err
```

Comment:
Why do we need to wrap the exception?

Reply:
This logs
> WARNING (MainThread) [homeassistant.components.automation] Invalid condition: Error rendering condition enabled template: TemplateError: Use of 'states' is not supported in limited templates

wheras bubbling up the `TemplateError` directly would not point to the `enabled:` key.
> WARNING (MainThread) [homeassistant.components.automation] Invalid condition: TemplateError: Use of 'states' is not supported in limited templates